### PR TITLE
Removed get_it

### PR DESCRIPTION
**Pull request by zenled**
Updated to `riverpod2` and removed `get_it`.

The main problem here was, in which file to declare the providers. 
I will present this on pub beer next week, but for now, I decided to put it next to the interface declaration.
(`@riverpod LoginInteractor loginInteractor(LoginInteractorRef ref) { ...` is above the `abstract class LoginInteractor`)

### CODE REVIEW

**Review by Fle9ma — APPROVED** [review]
Looks good to me! 😊 🙌🏻

File: lib/domain/data/user_data_holder.dart [hunk 1]
```diff
@@ -1,4 +1,12 @@
...
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'user_data_holder.g.dart';
+
+@Riverpod(keepAlive: true)
+UserDataHolder userDataHolder(UserDataHolderRef ref) {
+  return UserDataHolder();
+}
 
 class UserDataHolder {
```
Comment (Truba) [inline lib/domain/data/user_data_holder.dart:11]: Could have this just been, like a regular provide class?
```suggestion
@Riverpod(keepAlive: true)
class UserDataHolder {
```

File: lib/domain/interactor/dashboard/fetch_feed_interactor.dart [hunk 2]
```diff
@@ -1,4 +1,14 @@
...
+import 'package:flutter_dasher/domain/interactor/dashboard/fetch_feed_interactor_impl.dart';
+import 'package:flutter_dasher/domain/repository/feed_repository.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'fetch_feed_interactor.g.dart';
+
+@riverpod
+FetchFeedInteractor fetchFeedInteractor(FetchFeedInteractorRef ref) {
+  return FetchFeedInteractorImpl(ref.read(feedRepositoryProvider));
+}
```
Comment (Truba) [inline lib/domain/interactor/dashboard/fetch_feed_interactor.dart:11]: I'm perfectly fine with this being here. I would also be fine with code like this replacing the code in inject dependencies :)

File: lib/domain/data/user_data_holder.dart (hunk 1 above)
Comment (Truba) [inline lib/domain/data/user_data_holder.dart:11]: Could have this just been, like a regular provide class?
```suggestion
@Riverpod(keepAlive: true)
class UserDataHolder {
```

File: lib/ui/dashboard/presenter/current_user_presenter.dart [hunk 3]
```diff
@@ -1,8 +1,7 @@
 import 'package:flutter_dasher/common/model/user.dart';
 import 'package:flutter_dasher/domain/data/user_data_holder.dart';
-import 'package:get_it/get_it.dart';
 import 'package:hooks_riverpod/hooks_riverpod.dart';
 
 final currentUserPresenter = StateProvider.autoDispose<User>(
-  (ref) => GetIt.instance.get<UserDataHolder>().user!,
+  (ref) => ref.watch(userDataHolderProvider).user!,
 );
```
Comment (Truba) [inline lib/ui/dashboard/presenter/current_user_presenter.dart:7]: Should we make this one code gen as well?

Comment (Truba) [inline lib/ui/dashboard/presenter/current_user_presenter.dart:7]: Should we make this one code gen as well?

File: lib/ui/dashboard/presenter/feed_request_presenter.dart [hunk 4]
```diff
@@ -1,21 +1,20 @@
 import 'package:flutter_dasher/common/model/tweet.dart';
 import 'package:flutter_dasher/domain/interactor/dashboard/fetch_feed_interactor.dart';
 import 'package:flutter_dasher/ui/common/bits/request_provider/request_provider.dart';
-import 'package:get_it/get_it.dart';
 import 'package:hooks_riverpod/hooks_riverpod.dart';
 
 final feedRequestPresenter = ChangeNotifierProvider.autoDispose<FeedRequestPresenter>(
-  (ref) => FeedRequestPresenter(GetIt.instance.get()),
+  (ref) => FeedRequestPresenter(ref.read(fetchFeedInteractorProvider)),
 );
```
Comment (Truba) [inline lib/ui/dashboard/presenter/feed_request_presenter.dart:8]: Few more places like this where we can use code gen :)

File: lib/domain/data/user_data_holder.dart (hunk 1 above)
Comment (zenled) [reply lib/domain/data/user_data_holder.dart:11]: Sadly no. To use the `@riverpod` above the class, the class needs to be a `Notifier`:
```dart
@riverpod
class MyNotifier extends _$MyNotifier {
  @override
  Result build() {}
}
```

To provide regular classes object you must create a function:
```dart
@riverpod
RegularClass regularClass(regularClassRef ref) {
   return RegularClass();
}
```

File: lib/ui/dashboard/presenter/current_user_presenter.dart (hunk 3 above)
Comment (zenled) [reply lib/ui/dashboard/presenter/current_user_presenter.dart:7]: Not sure what you mean here.
I'm planning to make the `CurrentUserPresenter` a `Notifier` in the next PR.

Comment (Truba) [reply lib/ui/dashboard/presenter/current_user_presenter.dart:7]: Then I'll be eagerly awaiting the next PR 😊

Comment (Truba) [reply lib/ui/dashboard/presenter/current_user_presenter.dart:7]: Then I'll be eagerly awaiting the next PR 😊

**Pull request merged**